core-bundle/src/Twig/Loader/TemplateLocator.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Twig\Loader;
+
+use Contao\CoreBundle\HttpKernel\Bundle\ContaoModuleBundle;
+use Symfony\Component\Finder\Finder;
+use Webmozart\PathUtil\Path;
+
+class TemplateLocator
```

Comment:
This is functionally related to the `AddResourcesPass` but currently cannot reuse due to different requirements. We might want to revisit this (in the future) and try to homogenize it.